src/Compilers/Core/Portable/Diagnostic/TriggerDiagnosticDescriptor.cs

```diff
@@ -18,7 +18,7 @@ public class TriggerDiagnosticDescriptor : DiagnosticDescriptor
         /// <param name="customTags">Optional custom tags for the diagnostic. See <see cref="WellKnownDiagnosticTags"/> for some well known tags.</param>
         public TriggerDiagnosticDescriptor(string id, params string[] customTags)
             : base(id, title: "", messageFormat: "", category: "", defaultSeverity: DiagnosticSeverity.Hidden, isEnabledByDefault: true,
-                    customTags: customTags.Append(WellKnownDiagnosticTags.NotConfigurable))
+                    customTags: customTags.Append(WellKnownDiagnosticTags.NotConfigurable).Append(WellKnownDiagnosticTags.Trigger))
```

Comment:
if we are going to add Trigger custom tag, we should change SimpleDiagnostic.Create to create TriggerDiagnosticDescriptor rather than regular DescriptorDiagnostic.